src/utils.rs

```diff
@@ -0,0 +1,54 @@
+use crate::*;
+
+impl Dao {
+    // calculates the funds that the member can redeem based on his shares
+    pub async fn redeemable_funds(&self, share: u128) -> u128 {
+        let balance = balance(&self.approved_token_program_id, &exec::program_id()).await;
+        (share * balance) / self.total_shares
+    }
+
+    // checks that account is DAO member
+    pub fn is_member(&self, account: &ActorId) -> bool {
+        match self.members.get(account) {
+            Some(member) => {
+                if member.shares == 0 {
+                    return false;
+                }
+            }
+            None => {
+                return false;
+            }
+        }
+        true
```

Comment:
I think this condition may be simpler.
```suggestion
        matches!(self.members.get(account), Some(member) if member.shares != 0)
```